prusti-specs/src/lib.rs

```diff
@@ -238,6 +239,10 @@ fn generate_for_pure(attr: TokenStream, item: &untyped::AnyFnItem) -> GeneratedR
         ));
     }
 
+    do_generate_for_pure(item)
+}
+
+fn do_generate_for_pure(item: &untyped::AnyFnItem) -> GeneratedResult {
```

Comment:
This is not an amazing name, perhaps add a one-line comment explaining that this one is usable from other parts of `prusti-specs` and that it doesn't receive or check the attribute tokens?

Reply:
Good catch! In any other language, I would have just overloaded it, but coming up with a meaningful name is hard. Maybe we could incorporate the word `unchecked` somehow?

Reply:
Actually, what if we get rid of this and instead call `generate_spec_and_assertions` directly from `ghost_constraints::generate`? We'd could either remap `NestedSpec` to `SpecAttributeKind` or directly parse as the latter to get rid of the current (minor) duplicate parsing logic, though this would affect `|=` as well…